pkg/lib/operation/project/remote/job/run/operation.go

```diff
@@ -0,0 +1,184 @@
+package run
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/keboola/go-client/pkg/keboola"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type dependencies interface {
+	KeboolaProjectAPI() *keboola.API
+	ProjectFeatures() keboola.FeaturesMap
+	Logger() log.Logger
+	Tracer() trace.Tracer
+}
+
+type RunOptions struct {
+	Jobs    []Job
+	Async   bool
+	Timeout time.Duration
+}
+
+func Run(ctx context.Context, o RunOptions, d dependencies) (err error) {
+	ctx, span := d.Tracer().Start(ctx, "kac.lib.operation.project.remote.job.run")
+	defer telemetry.EndSpan(span, &err)
+
+	logger := d.Logger()
+	api := d.KeboolaProjectAPI()
+	timeoutCtx, cancel := context.WithTimeout(ctx, o.Timeout)
+	defer cancel()
+
+	queue := &JobQueue{
+		hasQueueV2:     d.ProjectFeatures().Has("queuev2"),
+		ctx:            timeoutCtx,
+		api:            api,
+		wg:             &sync.WaitGroup{},
+		err:            errors.NewMultiError(),
+		remaining:      map[string]bool{},
+		remainingMutex: &sync.Mutex{},
+		done:           make(chan struct{}),
+	}
+
+	if len(o.Jobs) > 1 {
+		logger.Infof("Starting %d jobs.", len(o.Jobs))
+	} else {
+		logger.Info("Starting job.")
+	}
+
+	// dispatch jobs in parallel
+	for _, job := range o.Jobs {
+		job := job
+		queue.dispatch(&job)
+	}
+
+	queue.startLogRemaining(logger)
+	err = queue.wait()
+	queue.stopLogRemaining()
+
+	if err != nil {
+		logger.Error("Some jobs failed, see below.")
+		return err
+	} else {
+		if !o.Async {
+			logger.Info("Finished running all jobs.")
+		} else {
+			logger.Info("Started all jobs.")
+		}
+		return nil
+	}
+}
+
+type JobQueue struct {
+	hasQueueV2 bool
+
+	ctx context.Context
+	api *keboola.API
+
+	wg  *sync.WaitGroup
+	err errors.MultiError
+
+	remaining      map[string]bool
+	remainingMutex *sync.Mutex
+
+	done chan struct{}
+}
+
+func (q *JobQueue) startLogRemaining(logger log.Logger) {
+	go func() {
+		for {
+			select {
+			case <-q.ctx.Done():
+				break
+			case <-q.done:
+				break
+			case <-time.After(time.Second * 5):
+				logger.Infof("Waiting for %s", strings.Join(q.getRemainingJobs(), ", "))
```

Comment:
![image](https://user-images.githubusercontent.com/19371734/218676158-bd7e4670-6ab9-4318-a116-40ebbebafb08.png)

- :large_blue_circle: super, ze si myslel na to, ked niekto spusti rovnaky config viac krat
  - len je to dost edge case, tu `(0)` by som tam nevypisoval, ono nie je velmi jasne z toho vystupu co to je.
- :purple_circle: V chybovej hlaske je ID jobu, ale ja neviem ku ktorej konfiguracii/komponente to patri.
  - ak by tam bolo `Job "34739670" from "keboola.wr-synapse/1025114" failed: ...` tak by to bolo super

Reply:
Plus, v tom `Waiting ...` by bolo super mat zase `jobID`, ze si to vies napr. v UI najst, ked to trva nejak dlho.

Reply:
Este jeden napad.

Mohol by si zalogovat, ze `Started job "34739670" from "keboola.wr-synapse/1025114".`

A vo waiting uz iba vypisovat `jobID`.

Reply:
Tym padom by si nemusel riesit ani tie `(0)/(1)` ....

Reply:
Upraveno následovně:

Async
![image](https://user-images.githubusercontent.com/1665677/218729553-01eebbbc-fb2a-4531-8a11-f28ba827251b.png)

Error
![image](https://user-images.githubusercontent.com/1665677/218729628-9c79fd6b-bae3-4226-b386-f9400e5ee052.png)

Success
![image](https://user-images.githubusercontent.com/1665677/218729817-657937e2-c941-4ff8-8cd8-ec8fc08c15a2.png)